envalid.d.ts

```diff
@@ -24,6 +24,11 @@ interface Spec<T> {
      * A url that leads to more detailed documentation about the env var.
      */
     docs?: string
+    /**
+     * A function that receives the full raw environment and returns a boolean
+     * indicating if this field is required or optional.
+     */
+    requiredWhen?: (env: any) => boolean
```

Comment:
Out of curiosity, why `any` instead of `NodeJS.ProcessEnv`? Speaking for the whole file here...

Reply:
[Looks like `NodeJS.ProcessEnv` will accept any indexable object where the key is a `string` and the value is either a `string` or `undefined`.](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/index.d.ts#L675-L677)

```typescript
declare namespace NodeJS {
        // ...
        export interface ProcessEnv {
                [key: string]: string | undefined;
        }
        // ...
}
```

So your example of `{ someFake: 'env' }` would compile, but

- `{ someFake: null }`
- `{ someFake: 42 }`
- `{ someFake: false }`
- `{ someFake: [1, 2, 3] }`
- `{ someFake: { really: 'fake' } }`
- `{ someFake: fake => anotherFake }`
- `{ someFake: Symbol('gimli') }`

... would not. This makes sense to me since env vars are read in to `process.env` as strings.

<p align="center">
  <img src="https://d.pr/f/9I3lhP+">
</p>

I think you'd probably get away with just changing `envalid.d.ts`, if you wanted to. 

Should make this change in a different PR if y'all wanted it; just seemed like a good place to ask. 

😉

Reply:
We've explicitly added support for `{ someFake: false }` etc for testing purposes, so they should keep working...

But yeah, feel free to open up a separate issue!